homeassistant/components/bmw_connected_drive/__init__.py

```diff
@@ -161,6 +161,17 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
+async def async_remove_config_entry_device(
```

Comment:
Is it not possible to clean up old devices automatically when setting up the config entry?

Reply:
That would be possible. However I'm not sure if this is desired from a user-perspective, as by doing so the old entity data would be deleted as well (assuming I understand it correctly)?

I've kept in line with two other integrations I use, namly fritzbox (which has this functionality) and deconz (which provides a seperate service for cleaning up).

Reply:
What would be the reason for keeping a stale device?

Reply:
Thought about it for a while and couldn't think of it any really valid reason. Adjusted the code to remove old vehicles automatically.